asg0_2.c

```diff
@@ -0,0 +1,144 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct record {
+    int id;
+    char name[100];
+    int age;
+};
+
+int isvalid(FILE *ptr,int id){
+    struct record user;
+    rewind(ptr);
+    while(fscanf(ptr,"%d %s %d\n",&user.id, user.name, &user.age)==3){
+        if(user.id==id){
+            printf("ID %d already exists.\n",id);
+            return 0;
+        }
+    }
+    return 1;
+}
+void create_op(FILE *ptr){
+    struct record user;
+    printf("Enter ID:");
+    scanf("%d", &user.id);
+    while(!isvalid(ptr,user.id)){
+        printf("Please enter a valid unique ID.\n");
+        printf("Enter ID:");
+        scanf("%d", &user.id);     
+    }
+    
+    printf("Enter Name:");
+    scanf("%s", user.name);
+    printf("Enter Age:");
+    scanf("%d", &user.age);
+
+    fprintf(ptr, "%d %s %d\n", user.id, user.name, user.age);
+    printf("User added successfully.\n");
+}
+
+
+void update_op(FILE *ptr, int user_id) {
+    struct record user;
+    FILE *temp = fopen("temp.txt", "w");
+    rewind(ptr);
+    int found = 0;
+    while (fscanf(ptr, "%d %s %d\n", &user.id, user.name, &user.age) != EOF) {
+        if (user.id == user_id) {
+            printf("Enter new Name and Age: ");
+            scanf("%s %d", user.name, &user.age);
+            found = 1;
+        }
+        fprintf(temp, "%d %s %d\n", user.id, user.name, user.age);
+    }
+    fclose(ptr);
+    fclose(temp);
+    remove("users.txt");
+    rename("temp.txt", "users.txt");

```

Comment:
can we have another better approach, instead of using extra file resource and having extra operations of removing and renaming?